applications/RANSApplication/python_scripts/coupled_rans_solver.py

```diff
@@ -0,0 +1,303 @@
+# Importing the Kratos Library
+import KratosMultiphysics as Kratos
+from KratosMultiphysics import IsDistributedRun
+from KratosMultiphysics.kratos_utilities import CheckIfApplicationsAvailable
+from KratosMultiphysics.python_solver import PythonSolver
+
+# Import applications
+import KratosMultiphysics.FluidDynamicsApplication as KratosCFD
+
+# Import application specific modules
+from KratosMultiphysics.RANSApplication.formulations import Factory as FomulationFactory
+from KratosMultiphysics.RANSApplication import RansVariableUtilities
+from KratosMultiphysics.FluidDynamicsApplication.check_and_prepare_model_process_fluid import CheckAndPrepareModelProcess
+
+# case specific imports
+if (IsDistributedRun() and CheckIfApplicationsAvailable("TrilinosApplication")):
+    import KratosMultiphysics.TrilinosApplication as KratosTrilinos     # MPI solvers
+    from KratosMultiphysics.mpi.distributed_import_model_part_utility import DistributedImportModelPartUtility
+elif (IsDistributedRun()):
+    raise Exception("Distributed run requires TrilinosApplication")
+
+class CoupledRANSSolver(PythonSolver):
```

Comment:
Two questions:
1) Why coupled?
2) Why not derived from the `FluidSolver`?

Reply:
1. Because this solver solves given any number of coupled transport equations via Formulations. (Any suggestions for naming?)
2. FluidSolver is very specific (it has some methods like `_GetBuilderAndSolver`, which in this case is handled by `Formulation`. This solver doesn't know what type of equations are being solved, Its sole responsibility is to prepare environment so that `Formulations` can do solving their respective transport equations.